Mathlib/RingTheory/MvPolynomial/BinaryForm.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Mathlib.RingTheory.MvPolynomial.Homogeneous
+
+/-!
+# Binary forms
+
+A binary form of degree `d` is a homogeneous polynomial in two variables of degree `d`.
+
+## Main definitions
+
+- `BinaryForm R d` is the type of binary forms over a commutative ring `R` of degree `d`.
+- `BinaryForm.basis R d` is the basis of `BinaryForm R d` consisting of the monomials
+  `X ^ i * Y ^ (d - i)` for `i : Fin (d + 1)`.
+
+-/
+
+
+universe u
+
+open MvPolynomial Finsupp Module
+
+/-- A binary form of degree `d` is a homogeneous polynomial in two variables of degree `d`. -/
+abbrev BinaryForm (R : Type u) [CommSemiring R] (d : ℕ) : Type u :=
+  homogeneousSubmodule (Fin 2) R d
+
+namespace BinaryForm
+
+variable (R : Type u) [CommSemiring R] (d : ℕ)
+
+/-- The bijection from `Fin (d + 1)` to degrees of bivariate monomials with degree `d`.
+`i : Fin (d + 1)` is sent to the degree of `X ^ i * Y ^ (d - i)`. -/
+@[simps!] noncomputable def degreeEquiv : Fin (d + 1) ≃ { c : Fin 2 →₀ ℕ // c.degree = d } where
```

Comment:
I'm not convinced this is the right lemma since it's not really about polynomials, it's redundant to talk about finitely-supported functions on a finite type, and it's very specific.

Perhaps instead of adding this, you could find / add lemmas along the lines of:
```lean
-- Maybe this exists somewhere (didn't look)
def foo : Fin (d + 1) ≃ {c : Fin 2 → ℕ // c 0 + c 1 = d} where
  toFun v := ⟨![v, d - v], by simp only [Fin.isValue, Matrix.cons_val]; omega⟩
  invFun c := Fin.ofNat _ (c.1 0)
  left_inv v := by simp
  right_inv := sorry -- etc
```
and:
```lean
-- Relation of `Finsupp.degree` and `Finsupp.equivFunOnFinite` (hopefully exists already)
def bar {ι : Type*} [Fintype ι] (n : ℕ) :
    {f : ι →₀ ℕ // f.degree = n} ≃ {f : ι → ℕ // ∑ i, f i = n} := by
  sorry
```